File: OneCodeTeam/How to post data with HttpWebRequest in ASP.NET/[VB]-How to post data with HttpWebRequest in ASP.NET/VB/VBASPNETHttpWebRequest/WCFRESTService/RESTAPI.svc.vb
'***************************** Module Header ******************************\
' Module Name:	IRESTAPI.svc.vb
' Project:		WCFRESTService
' Copyright (c) Microsoft Corporation.
' 
' This sample will show you how to create HTTPWebReqeust, and get HTTPWebResponse.
'
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'**************************************************************************/

' NOTE: You can use the "Rename" command on the context menu to change the class name "RESTAPI" in code, svc and config file together.
' NOTE: In order to launch WCF Test Client for testing this service, please select RESTAPI.svc or RESTAPI.svc.vb at the Solution Explorer and start debugging.
Public Class RESTAPI
    Implements IRESTAPI

    Public Function GetUserNameByID(id As String) As String Implements IRESTAPI.GetUserNameByID
        Try
            Dim userID As Integer = Integer.Parse(id)
            If userID > 0 AndAlso userID < 5 Then
                Dim userDataList = (New UserDataList()).getUserDataList()
                Dim query = From data In userDataList Where data.ID = id Select data.UserName
                Return query.FirstOrDefault()
            Else
                Return "Can't find the user with id: " & id
            End If
        Catch e As Exception
            Throw New Exception(String.Format("Can't convert {0} to int", id))
        End Try
    End Function

    Public Function GetUserNameByPostID(rData As UserData) As UserData Implements IRESTAPI.GetUserNameByPostID
        Try
            Dim userID As Integer = Integer.Parse(rData.ID)
            If userID > 0 AndAlso userID < 5 Then
                Dim userDataList = (New UserDataList()).getUserDataList()
                Dim query = From data In userDataList Where data.ID = rData.ID Select data
                Return New UserData() With { _
                     .UserName = query.FirstOrDefault().UserName _
                }
            Else
                Return Nothing
            End If
        Catch
            Throw New Exception(String.Format("Can't convert {0} to int", rData.ID))
        End Try


    End Function
End Class

File: OneCodeTeam/How to post data with HttpWebRequest in ASP.NET/[C#]-How to post data with HttpWebRequest in ASP.NET/C#/CSASPNETHttpWebRequest/WCFRESTService/RESTAPI.svc.cs
/***************************** Module Header ******************************\
* Module Name:	RESTAPI.svc.cs
* Project:		WCFRESTService
* Copyright (c) Microsoft Corporation.
* 
* This sample will show you how to create HTTPWebReqeust, and get HTTPWebResponse.
*
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\**************************************************************************/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.Text;

namespace WCFRESTService
{
    
    public class RESTAPI : IRESTAPI
    {
        public string GetUserNameByID(string id)
        {     
            try 
	        {
                int userID = int.Parse(id);
                if (userID > 0 && userID < 5)
                {
                    var userDataList = (new UserDataList()).getUserDataList();
                    var query = from data in userDataList
                                where data.ID == id
                                select data.UserName;
                    return query.FirstOrDefault();
                }
                else
                {
                    return "Can't find the user with id: " + id;
                }        
	        }
	        catch 
	        {
                throw new Exception(string.Format("Can't convert {0} to int", id));
	        }
        }

        public UserData GetUserNameByPostID(UserData rData)
        {
            try
            {
                int userID = int.Parse(rData.ID);
                if (userID > 0 && userID < 5)
                {
                    var userDataList = (new UserDataList()).getUserDataList();
                    var query = from data in userDataList
                                where data.ID == rData.ID
                                select data;
                    return new UserData { UserName = query.FirstOrDefault().UserName };
                }
                else
                {
                    return null;
                }
            }
            catch
            {
                throw new Exception(string.Format("Can't convert {0} to int", rData.ID));
            }


        }
    }
}
